nucypher/blockchain/eth/sol/compile/solc.py

```diff
@@ -0,0 +1,58 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+from logging import Logger
+from typing import Dict, Optional, List
+
+from nucypher.blockchain.eth.sol.compile.config import OPTIMIZER_RUNS
+from nucypher.blockchain.eth.sol.compile.constants import SOLC_LOGGER, SOLIDITY_SOURCE_ROOT, TEST_SOLIDITY_SOURCE_ROOT
+from nucypher.blockchain.eth.sol.compile.exceptions import CompilationError
+from nucypher.blockchain.eth.sol.compile.types import VersionString
+from nucypher.exceptions import DevelopmentInstallationRequired
+
+
+def __execute(compiler_version: VersionString, input_config: Dict, allow_paths: Optional[List[str]] = None):
+    """Executes the solcx command and underlying solc wrapper"""
+    log = Logger('execute-solcx')
+
+    # Lazy import to allow for optional installation of solcx
+    try:
+        from solcx.install import get_executable
+        from solcx.main import compile_standard
+    except ImportError:
+        raise DevelopmentInstallationRequired(importable_name='solcx')
+
+    # Prepare Solc Command
+    solc_binary_path: str = get_executable(version=compiler_version)
+    SOLC_LOGGER.info(f"Compiling with base path")  # TODO: Add base path
+
+    _allow_paths = f'{SOLIDITY_SOURCE_ROOT},{TEST_SOLIDITY_SOURCE_ROOT}'
+    if allow_paths:
+        _allow_paths += ','.join(str(p) for p in allow_paths)
```

Comment:
allow paths will always come from source bundle (or at least should), so how about to remove constants from here?